Add io.gitlab.JBeaumont81.Etesync-Knotes

JBeaumont81 opened the pull request:
<!-- ⚠️⚠️  Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Please go to the preview tab to view the markdown below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

<!-- 💡 Please tick and write 'N/A' with a reason if a checklist item below is not applicable 💡 -->

- [x] Please describe the application briefly. A secure note-taking application for linux, integrated with Etebase for encrypted synchronization.
- [x] Please attach a video showcasing the application on Linux using the Flatpak. https://gitlab.com/JBeaumont81/Etesync-Knotes/-/blob/main/Etesync-Knotes.demo.mp4
- [x] The Flatpak ID follows all the rules listed in the [Application ID requirements][appid].
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [X] I am an author/developer of the project.

<!-- 💡 Please mention below the GitHub usernames of any additional maintainers needed (if any) 💡 -->

<!-- ⚠️⚠️  Please DO NOT modify anything below this line ⚠️⚠️  -->

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

bbhtt commented on io.gitlab.JBeaumont81.EtesyncKnotes.yaml:
```diff
@@ -0,0 +1,45 @@
+app-id: io.gitlab.JBeaumont81.EtesyncKnotes
+runtime: org.kde.Platform
+runtime-version: '6.7'
```
```suggestion
runtime-version: '6.9'
```

JBeaumont81 replied:
updated

bbhtt commented on io.gitlab.JBeaumont81.EtesyncKnotes.yaml:
```diff
@@ -0,0 +1,45 @@
+runtime-version: '6.7'
+sdk: org.kde.Sdk//6.7
+command: etesync-knotes
+separate-locales: false
```
why?

JBeaumont81 replied:
Why would I need to separate the locales? it will increase the size of the flatpak.

JBeaumont81 replied:
Ok thanks for the clarification. I reverted  back to default.

bbhtt commented on io.gitlab.JBeaumont81.EtesyncKnotes.yaml:
```diff
@@ -0,0 +1,45 @@
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=ipc
+  - --share=network
```
why is network needed?

bbhtt commented on io.gitlab.JBeaumont81.EtesyncKnotes.yaml:
```diff
@@ -0,0 +1,45 @@
+  - --share=ipc
+  - --share=network
+  - --filesystem=home
+  - --talk-name=org.kde.kwalletd6
```
what is this for?

bbhtt commented on io.gitlab.JBeaumont81.EtesyncKnotes.yaml:
```diff
@@ -0,0 +1,45 @@
+
+modules:
+  - name: libetebase
+    buildsystem: simple
+    build-commands:
+      # Install .so from Git path
+      - install -Dm755 libetebase/lib/libetebase.so "${FLATPAK_DEST}/lib/libetebase.so"
+      - ln -sf "${FLATPAK_DEST}/lib/libetebase.so" "${FLATPAK_DEST}/lib/libetebase.so.0"
+      # Install headers from Git path
+      - install -Dm644 -t "${FLATPAK_DEST}/include/etebase" libetebase/include/etebase/*.h
+    sources:
+      - type: git
+        url: https://gitlab.com/JBeaumont81/Etesync-Knotes.git
+        tag: v1.0f
```
why did you commit shared libs in git repo? Build it from source here

JBeaumont81 replied:
What would be acceptable ? do you have any suggestion?
I have no control over how libetebase is developed which I think is the root cause of this issue.

bbhtt replied:
open an issue with them and discuss your problems, not me or don't use that lib.

sticking a random lib in git repo is an issue.

JBeaumont81 replied:
For info, for the libetebase module, if I don't use:

`build-options:
  build-args:
    - --share=network`

Here are the errors I got:

```
Running: . /usr/lib/sdk/rust-stable/enable.sh && cargo build --release
    Updating crates.io index
warning: spurious network error (3 tries remaining): [6] Could not resolve hostname (Could not resolve host: index.crates.io)
warning: spurious network error (2 tries remaining): [6] Could not resolve hostname (Could not resolve host: index.crates.io)
warning: spurious network error (1 tries remaining): [6] Could not resolve hostname (Could not resolve host: index.crates.io)
error: failed to get `etebase` as a dependency of package `libetebase v0.5.8 (/run/build/libetebase)`
```

hfiguiere replied:
https://docs.flathub.org/docs/for-app-authors/requirements#dependency-manifest

JBeaumont81 commented on io.gitlab.JBeaumont81.EtesyncKnotes.yaml:
```diff
@@ -0,0 +1,45 @@
+app-id: io.gitlab.JBeaumont81.EtesyncKnotes
+runtime: org.kde.Platform
+runtime-version: '6.7'
+sdk: org.kde.Sdk//6.7
```
updated

bbhtt commented on app/share/metainfo/io.gitlab.JBeaumont81.Etesync-Knotes.metainfo.xml:
this file should not be here.

JBeaumont81 replied:
Sorry I'm new to this. That what I understood from the MetaInfo guidellines. Where should it be?

bbhtt replied:
in your upstream repo

JBeaumont81 replied:
Moved to https://gitlab.com/JBeaumont81/Etesync-Knotes

bbhtt commented on flathub.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "skip-arches": ["aarch64"]
+}
```
not acceptable.

JBeaumont81 replied:
file remove. I don't know if it will work on aarch64. but now libetebase should compile from the source.

JBeaumont81 replied:
Desolé, I went too fast yesterday. Gone

JBeaumont81 commented on io.gitlab.JBeaumont81.EtesyncKnotes.yaml:
```diff
@@ -0,0 +1,45 @@
+  - --talk-name=org.kde.kwalletd6
+  - --device=dri
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --env=QT_QPA_PLATFORM=wayland;xcb
```
I removed :
  - --device=dri
  - --talk-name=org.kde.StatusNotifierWatcher

bbhtt replied:
> I removed : - --device=dri - --talk-name=org.kde.StatusNotifierWatcher

and where did i suggest that?

JBeaumont81 replied:
I reverted to original code

bbhtt replied:
the suggestion is still not applied.

bbhtt replied:
no

bbhtt commented on io.gitlab.JBeaumont81.EtesyncKnotes.yaml:
```diff
@@ -0,0 +1,45 @@
+  - --socket=wayland
+  - --share=ipc
+  - --share=network
+  - --filesystem=home
```
still wrong, use XDG_CONFIG_HOME.

JBeaumont81 replied:
Done

hfiguiere commented:
The checklist is missing (ie it was removed by the chatbot)

JBeaumont81 commented on io.gitlab.JBeaumont81.Etesync-Knotes.yaml:
```diff
@@ -0,0 +1,1963 @@
+  - name: etesync-knotes
+    buildsystem: simple
+    build-commands:
+      - cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/app -DCMAKE_INCLUDE_PATH=/app/include -DCMAKE_LIBRARY_PATH=/app/lib
```
simple allows the 3  manual "install"  commands to run. I understand that cmake-ninja would end-up run make install instead (which is not define).

JBeaumont81 commented on Cargo.lock:
My bad. Removed

JBeaumont81 commented on io.gitlab.JBeaumont81.Etesync-Knotes.yaml:
```diff
@@ -0,0 +1,1963 @@
+      - type: file
+        path: Cargo.lock
+        dest-filename: Cargo.lock  # Explicitly set this to avoid the error
+      - type: inline
```
No sure I got this right. I removed the reference to the cargo.lock file and made a reference to the cargo-sources.json rather than having everything in the manifest. The faltpak gets build properly that way. Was it what you were referring to?

bbhtt replied:
it is correct now.

bbhtt commented on io.gitlab.JBeaumont81.Etesync-Knotes.yaml:
```diff
@@ -0,0 +1,54 @@
+    no-make-install: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCMAKE_INSTALL_PREFIX=/app
+      - -DCMAKE_INCLUDE_PATH=/app/include
+      - -DCMAKE_LIBRARY_PATH=/app/lib
```
```suggestion
```

bbhtt commented on io.gitlab.JBeaumont81.Etesync-Knotes.yaml:
```diff
@@ -0,0 +1,54 @@
+      - -DCMAKE_INSTALL_PREFIX=/app
+      - -DCMAKE_INCLUDE_PATH=/app/include
+      - -DCMAKE_LIBRARY_PATH=/app/lib
+    post-install:
+      - install -Dm755 etesync-knotes "${FLATPAK_DEST}/bin/etesync-knotes"
+      - install -Dm644 desktop/etesync-knotes.desktop "${FLATPAK_DEST}/share/applications/etesync-knotes.desktop"
+      - install -Dm644 icons/etesync-knotes.svg "${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/etesync-knotes.svg"
```
why isn't this done by your cmake?

JBeaumont81 replied:
I've been using Makefile for it

hfiguiere replied:
why?

hfiguiere commented:
Except that the failure isn't aarch64

It's 
```
/bin/sh: line 1: /usr/lib/sdk/rust-stable/enable.sh: No such file or directory
```

Did you even try to build the flatpak locally?

JBeaumont81 commented:
> Did you even try to build the flatpak locally?

Yes I did run locally both architectrue and it was explained in the comment that I just deleted (I guess that's why you have seen it)
I found the reason why it would not compile locally on aarch64  yet it was building well on amd64. it was the same reason the build failed here with amd64. I fixed the manifest accordingly. I'm still having issue further down the road. libetebase compile fine but etesync-knotes fails. I'll run the bot in a bit to see if it's an issue only my side (maybe with the emulation).

JBeaumont81 commented:
bot, build

hfiguiere commented:
not changing anything will not solve the issue.

there is no appstream file installed.

JBeaumont81 commented:
In my defense I did change the tile of this PR from io.gitlab.JBeaumont81.EtesyncKnotes to io.gitlab.JBeaumont81.Etesync-Knotes, but I had little hope this was the issue

hfiguiere commented:
the title of the PR has no impact on the build.

JBeaumont81 commented:
bot, build

hfiguiere commented on io.gitlab.JBeaumont81.Etesync-Knotes.yaml:
```diff
@@ -0,0 +1,54 @@
+    sources:
+      - type: git
+        url: https://gitlab.com/JBeaumont81/Etesync-Knotes.git
+        tag: v1.0f
```
add a commit here as well.

hfiguiere commented on io.gitlab.JBeaumont81.Etesync-Knotes.yaml:
```diff
@@ -0,0 +1,54 @@
+    sources:
+      - type: git
+        url: https://github.com/etesync/libetebase.git
+        tag: v0.5.8
```
add a commit here as well.

bbhtt commented:
bot, build

bbhtt commented on io.gitlab.JBeaumont81.Etesync-Knotes.yaml:
```diff
@@ -0,0 +1,56 @@
+  - --socket=wayland
+  - --share=ipc
+  - --share=network
+  - --talk-name=org.kde.kwalletd6
```
does it only support kwallet? this won't work on anything that uses `org.freedesktop.secrets`.

hfiguiere commented on io.gitlab.JBeaumont81.Etesync-Knotes.yaml:
```diff
@@ -66,4 +66,4 @@ modules:
         tag: main
         tag: v1.1
-        commit: b3ddb2a256dd4244f922fc25423dd65cbd5c69cc
+        commit: ba3a6639402c9c2c695e47e0933d87fc439e1817
```
1. retagging is BAD practice.
2. you have two `tag` one being a branch.

hfiguiere commented:
bot, build

JBeaumont81 commented:
The flatpak from the test repo works as expect. Anything else needs to be done on my side?

bbhtt commented:
/merge head=a6960b8c652f774df99b13c8d47929bec0622874
